From f1ee090193ab67929506e2d71347bba67bbabf1f Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@gmail.com>
Date: Sat, 10 May 2014 22:18:36 +0100
Subject: [PATCH 1/6] build: Skip auto binary deps if IgnoreAutodep is set in
 actions.py

This is required in some cases where we encounter vicious circular
dependencies.. such as dbus/systemd. In these instances we'll not
bother trying to add automatic dependencies as we risk breaking
the world (TM)
---
 pisi/operations/build.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/pisi/operations/build.py b/pisi/operations/build.py
index e2862e6..af44279 100644
--- a/pisi/operations/build.py
+++ b/pisi/operations/build.py
@@ -1002,6 +1002,8 @@ class Builder:
                     metadata.package.providesPkgConfig.append(pkgconfig)
                     ctx.ui.debug(_("Adding %s to provided pkgconfig list" % pcName))
             else:
+                if self.actionGlobals.get("IgnoreAutodep"):
+                    continue
                 # Note this is quite slow to complete, but the only other
                 # alternative is exporting ELF symbols to the XML and then
                 # searching providers that way.. ala RPM.
-- 
1.9.1

